src/db_tables/user.py

```diff
@@ -0,0 +1,18 @@
+from typing import TYPE_CHECKING
+
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from src.utils.database import Base
+
+if TYPE_CHECKING:
+    from src.db_tables.user_list import UserList
```

Comment:
Add the explainer comment from readme for why these need to be in `TYPE_CHECKING`. (Applies for all occurences)
```suggestion
# Prevent circular imports for relationships
if TYPE_CHECKING:
    from src.db_tables.user_list import UserList
```